### PR TITLE
Fixed Executors/Stages tabs in Spark UI

### PR DESCRIPTION
**Pull request by szczeles**
Spark UI has a custom code to handle JS-capable tabs in SparkUI that doesn't like text "proxy" in the URL (see https://github.com/jupyterhub/jupyter-server-proxy/issues/57).

This PR injects one of the suggested solutions for this issue: https://github.com/jupyterhub/jupyter-server-proxy/issues/57#issuecomment-759388944, patching JS files in place

### CODE REVIEW

**Comment by szczeles** [comment]
Tested manually by running the image on Kubeflow
image

File: jupyterlab-mlops/Dockerfile [hunk 1]
```diff
@@ -19,7 +34,8 @@ USER jovyan
 RUN pip install --no-cache-dir jupyterlab-git
 
 # add kfp and install mlflow inside jupyterlab
-RUN mamba install --quiet --yes kfp mlflow=$MLFLOW_VERSION jupyter-server-proxy -c conda-forge && \
+RUN mamba install --quiet --yes kfp jupyter-server-proxy -c conda-forge && \
+    pip install mlflow==$MLFLOW_VERSION && \
```
Comment (marrrcin) [inline jupyterlab-mlops/Dockerfile:38]: `--no-cache-dir`?

File: jupyterlab-mlops/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
+    apt install -y zip patch && \
+    mkdir -p /tmp/patches/org/apache/spark/ui/static/ && \
+    unzip -p /usr/local/spark/jars/spark-core_2.12-3.1.2.jar org/apache/spark/ui/static/stagepage.js > /tmp/patches/org/apache/spark/ui/static/stagepage.js && \
+    unzip -p /usr/local/spark/jars/spark-core_2.12-3.1.2.jar org/apache/spark/ui/static/utils.js > /tmp/patches/org/apache/spark/ui/static/utils.js && \
+    patch /tmp/patches/org/apache/spark/ui/static/stagepage.js < /tmp/patches/stagepage.js.patch && \
+    patch /tmp/patches/org/apache/spark/ui/static/utils.js < /tmp/patches/utils.js.patch && \
+    pushd /tmp/patches && \
+    zip -u /usr/local/spark/jars/spark-core_2.12-3.1.2.jar org/apache/spark/ui/static/* && \
+    popd && \
+    rm -rf /tmp/patches
```
Comment (marrrcin) [inline jupyterlab-mlops/Dockerfile:21]: No cleanup after apt?

File: jupyterlab-mlops/Dockerfile (hunk 1 above)
Comment (szczeles) [reply jupyterlab-mlops/Dockerfile]: Good point, just added

File: jupyterlab-mlops/Dockerfile (hunk 2 above)
Comment (szczeles) [reply jupyterlab-mlops/Dockerfile]: Completely forgot about this :see_no_evil: Added now, thanks!

**Pull request merged**